dojo/tools/aws_inspector2/parser.py

```diff
@@ -0,0 +1,281 @@
+                finding.sast_source_file_path = f"{file_path}{file_name}"
+                finding.line = start_line
+                finding.sast_source_line = start_line
+                description += "**Additional info**\n"
+                description += f"CWEs: {', '.join(cwes)}\n"
+                description += (
+                    f"Vulnerability info from: {detector_id} {detector_name}\n"
+                )
+                description += f"Rule: {rule_id}\n"
+                description += f"Tags: {detector_tags}\n"
+                description += f"Lines: {start_line} - {end_line}\n"
+                description += f"URLs: {', '.join(reference_urls)}\n"
+                description += f"Lambda layer ARN: {layer_arn}\n"
+                finding.description = description
+
+            elif finding_type == "NETWORK_REACHABILITY":
+                # extract fields
+                network_path_info = raw_finding.get("networkPath", {})
+                network_path_steps = network_path_info.get("steps", [])
+                steps_descriptions = []
+                for step_number, step in enumerate(network_path_steps):
+                    steps_description = "steps:\n"
+                    component_id = step.get("componentId", "N/A")
+                    component_type = step.get("componentType", "N/A")
+                    steps_description += (
+                        f"{step_number}: {component_id} {component_type}"
+                    )
+                    steps_descriptions.append(steps_description)
+                open_port_range_info = raw_finding.get("openPortRange", {})
+                port_range_start = open_port_range_info.get("begin", "N/A")
+                port_range_end = open_port_range_info.get("end", "N/A")
+                protocol = raw_finding.get("protocol", "N/A")
+                # populate fields
+                description += "**Additional info**\n"
+                description += f"protocol {protocol}, port range {port_range_start} - {port_range_end}"
+                description += "\n".join(steps_descriptions)
+                description += "\n"
+            else:
+                msg = "Incorrect Inspector2 report format"
+                raise TypeError(msg)
+            # extract endpoints
+            # AWS_EC2_INSTANCE | AWS_ECR_CONTAINER_IMAGE | AWS_ECR_REPOSITORY | AWS_LAMBDA_FUNCTION
+            resource_infos = raw_finding.get("resources", {})
+            impact = []
+            endpoints = []
+            for resource_info in resource_infos:
+                resource_type = resource_info.get("type", None)
+                resource_id = resource_info.get("id", "N/A")
+                resource_details = resource_info.get("details", {})
+                resource_region = resource_info.get("region", "N/A")
+                endpoint_host = f"{resource_type} - {resource_id}"
+                if resource_type == "AWS_EC2_INSTANCE":
```

Comment:
It would make the code much easier to read by converting individual if cases into a function to process each resource type. This is